components/camera/fake/camera.go

```diff
@@ -176,53 +174,46 @@ type Camera struct {
 	Height int
 }
 
-// Read always returns the same image of a chess board.
+// Read always returns the same image of a yellow to blue gradient.
 func (c *Camera) Read(ctx context.Context) (image.Image, func(), error) {
-	img, err := c.getColorImage()
-	return img, func() {}, err
-}
-
-// NextPointCloud always returns a pointcloud of the chess board.
-func (c *Camera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
-	img, err := c.getColorImage()
-	if err != nil {
-		return nil, err
-	}
-	dm, err := c.getDepthImage(ctx)
-	if err != nil {
-		return nil, err
+	width := float64(c.Width)
+	height := float64(c.Height)
+	img := image.NewRGBA(image.Rect(0, 0, c.Width, c.Height))
+
+	totalDist := math.Sqrt(math.Pow(0-width, 2) + math.Pow(0-height, 2))
+
+	var x, y float64
+	for x = 0; x < width; x++ {
+		for y = 0; y < height; y++ {
+			dist := math.Sqrt(math.Pow(0-x, 2) + math.Pow(0-y, 2))
+			dist /= totalDist
+			thisColor := color.RGBA{uint8(255 - (255 * dist)), uint8(255 - (255 * dist)), uint8(0 + (255 * dist)), 255}
+			img.Set(int(x), int(y), thisColor)
+		}
```

Comment:
I think you can cache this generated image (and the pointcloud too). In the Camera struct, you can add a "cacheImage" and "cachePointCloud", that if the pointers are null, you can run this loop, but afterwards store the created image in memory, and just return that after the first time.